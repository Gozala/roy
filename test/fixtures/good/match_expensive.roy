data Foo = Bar | Baz

let expensive n count =
  if n == 0 then
    console.log "expensive called " ++ (String count) ++ " times."
    Bar()
  else
    expensive (n - 1) (count + 1)

let foo = match (expensive 10000 0)
  case Bar = true
  case Baz = false
