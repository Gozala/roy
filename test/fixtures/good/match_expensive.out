expensive called 10000 times.
